lib/orm/models.js

```diff
@@ -28,39 +28,88 @@ function createAssociationOptionsForFields(fields) {
           isInterfaceType(unwrappedType) || isUnionType(unwrappedType),
       };
 
-      associationOptions[fieldName] = isList
-        ? hasMany(associationName, options)
-        : belongsTo(associationName, options);
+      associations.push({
+        name: associationName,
+        fieldName: fieldName,
+        type: isList ? "hasMany" : "belongsTo",
+        options,
+      });
     }
 
-    return associationOptions;
+    return associations;
   };
 }
 
-function createAssociationOptions(type) {
+function createAssociations(type) {
   const fields = type.getFields();
-  const associationOptions = Object.keys(fields).reduce(
-    createAssociationOptionsForFields(fields),
-    {}
-  );
+  return Object.keys(fields).reduce(createAssociationForFields(fields), []);
+}
 
-  return associationOptions;
+function createModel(type) {
+  return {
+    name: type.name,
+    camelizedName: camelize(type.name),
+    associations: createAssociations(type),
+  };
 }
 
-function ensureModel(mirageSchema, type) {
-  const associationOptions = createAssociationOptions(type);
-  const model = Model.extend(associationOptions);
+function shouldCreateModel(type) {
+  return (
+    isObjectType(type) && !isRelayType(type) && !type.name.startsWith("__")
+  );
+}
 
-  mirageSchema.registerModel(type.name, model);
+function shouldRegisterModel(mirageSchema, name) {
+  return !mirageSchema.hasModelForModelName(name);
 }
 
-function shouldAddModel(mirageSchema, type) {
-  return (
-    !mirageSchema.hasModelForModelName(type.name) &&
-    isObjectType(type) &&
-    !isRelayType(type) &&
-    !type.name.startsWith("__")
+function registerModel(mirageSchema, model) {
+  const associationsOptions = model.associations.reduce(
+    (options, association) => {
+      options[association.fieldName] =
+        association.type === "hasMany"
+          ? hasMany(association.name, association.options)
+          : belongsTo(association.name, association.options);
+
+      return options;
+    },
+    {}
```

Comment:
Can I ask for 2 small changes here?

1. Can we extract the association options code to a separate function?
2. Super tiny nitpick... the code style in the project is to avoid arrow functions except for one-liners (or if we need to bind context but I don't think there are any cases of this). Do you mind replacing the one or two instances of multi-line arrow functions in this file with old school `function () {}` syntax to help with the consistency?